src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableOrchestrationClient.cs

```diff
@@ -267,6 +330,28 @@ async Task<IList<DurableOrchestrationStatus>> IDurableOrchestrationClient.GetSta
             return results;
         }
 
+        /// <inheritdoc />
+        async Task<T> IDurableOrchestrationClient.ReadActorState<T>(ActorId actorId, JsonSerializerSettings settings)
+        {
+            // TODO this cast is to avoid to change DurableTask.Core. Change it to use TaskHubClient.
+            var storageService = (AzureStorageOrchestrationService)this.client.ServiceClient;
+            var instanceId = ActorId.GetSchedulerIdFromActorId(actorId);
+            IList<OrchestrationState> stateList = await storageService.GetOrchestrationStateAsync(instanceId, false);
+
+            OrchestrationState state = stateList?.FirstOrDefault();
+            if (state == null
+                || state.OrchestrationInstance == null
+                || string.IsNullOrEmpty(state.Status)
+                || state.Status.StartsWith("Large"))
```

Comment:
The current implementation is quite simple: it just mirrors the actor state in the orchestration status as long as it is less than 16KB (because that seems to be a limitation on the orchestration status). This is meant really just for allowing efficient fast queries from the client that can to a simple point query in Azure storage. The "true" actor state remains in SchedulerState and can be any size.

I agree that it would probably be better to not just return null. Perhaps we can fall back onto a slower path that retrieves the larger state accurately.

Reply:
I am changing the implementation so it no longer uses the "Status" column and thus no longer has a size limitation. I am also adding a test to test that what I just said is actually true.